src/frontend/stmclient.cc

```diff
@@ -126,7 +127,7 @@ void STMClient::init( void )
   if ( (escape_key_env = getenv( "MOSH_ESCAPE_KEY" )) != NULL ) {
     if ( strlen( escape_key_env ) == 1 ) {
       escape_key = (int)escape_key_env[0];
-      if ( escape_key > 0 && escape_key < 128 ) {
+      if ( escape_key > 0 || escape_key < 128 ) {
```

Comment:
Good catch, @luto. What purpose this serves is a question for @rinne, the patch's original author. Blame says it comes from a49aef81.  Might be a mis-merge?